tests/integration/agents/supervisor/test_planner.py

```diff
@@ -0,0 +1,191 @@
+                ),
+                HumanMessage(
+                    content="Create a hello world app and deploy it with Kyma?"
+                ),
+            ],
+            '{"subtasks": [{"description": "Create a hello world app", "assigned_to": "Common"},'
+            '{"description": "deploy it with Kyma", "assigned_to": "KymaAgent"}]}',
+            False,
+        ),
+        (
+            # tests if a query related to Kyma and Common is divided into the correct subtasks for both agents
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(
+                    content="Create a hello world app with python and deploy it with Kyma?"
+                ),
+            ],
+            '{"subtasks": [{"description": "Create a hello world app with python", "assigned_to": "Common"},'
+            '{"description": "deploy it with Kyma", "assigned_to": "KymaAgent"}]}',
+            False,
+        ),
+        (
+            # tests if a complex query related to Kyma is divided correctly into two subtasks for the Kyma agent
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(
+                    content="how to enable eventing module and create a subscription for my app?"
+                ),
+            ],
+            '{"subtasks": [{"description": "How to enable eventing module?", "assigned_to": "KymaAgent"},'
+            '{"description": "How to create a subscription for my app?", "assigned_to": "KymaAgent"}]}',
+            False,
+        ),
+    ],
+)
+def test_invoke_planner(
+    messages,
+    expected_answer,
+    general_query,
+    companion_graph,
+    planner_correctness_metric,
+    answer_relevancy_metric,
+):
+    """Tests the invoke_planner method of SupervisorAgent."""
+    # Given: A conversation state with messages
+    state = create_mock_state(messages)
+
+    # When: The supervisor agent's planner is invoked
+    result = companion_graph.supervisor_agent._invoke_planner(state)
+    test_case = LLMTestCase(
+        input=messages[-1].content,
+        actual_output=result.content,
+        expected_output=expected_answer,
+    )
+
+    # Then: We evaluate based on query type
+    if not general_query:
+        # For specific queries, check planner correctness
+        planner_correctness_metric.measure(test_case)
+
+        print(f"Score: {planner_correctness_metric.score}")
+        print(f"Reason: {planner_correctness_metric.reason}")
+
+        companion_graph.plan_parser.parse(result.content)
+
+        assert_test(test_case, [planner_correctness_metric])
+    else:
+        # For general queries, check answer relevancy
```

Comment:
maybe more details what answer relevancy is? Does it compare gotten and expected answers?